tests/models/rwkv/test_modeling_rwkv.py

```diff
@@ -269,7 +269,10 @@ class RwkvModelTest(ModelTesterMixin, GenerationTesterMixin, PipelineTesterMixin
     pipeline_model_mapping = (
         {"feature-extraction": RwkvModel, "text-generation": RwkvForCausalLM} if is_torch_available() else {}
     )
-    # all_generative_model_classes = (RwkvForCausalLM,) if is_torch_available() else ()
+    all_generative_model_classes = (RwkvForCausalLM,) if is_torch_available() else ()
+    # Attention input support is not implemented; attention output has model-specific shapes (bsz, seq_len,
+    # hidden_size), as opposed to (bsz, num_heads, seq_len, embed_size_per_head), and has custom tests (see below)
+    has_attentions = False
```

Comment:
hmm - I don't think we should do this as this is a "global" attribute used to control many other tests which assume (correctly) that `has_attentions=False` indicates the model doesn't use attention or return attentions. Even if there's custom tests below - are we sure all of the tests which use `has_attentions` are overridden here? It's also not forwards compatible: I would want new attention tests added in test_modeling_common to fail and then require a custom implementation here, rather than be skipped. 

Instead, we'll need either a different argument OR doing something like this: 

```
def test_my_generative_test_foo():
    # comment explaining why
    old_has_attentions = self.has_attentions
    self.has_attentions = False
    super().test_my_generative_test_foo()
    self.has_attentions = old_has_attentions
```